### ISSUE TITLE
Trim Appended Filename

### ISSUE THREAD

**Issue by kwelch** [kwelch/babel-plugin-captains-log#17]
# What 
The file names appended to console statement include the entire path from root

## Example
`/Users/welchk/_git/enga-web_new/packages/client/src/App/events/Views/Catalog/index.js(189:4) filters []`

## Expected
`index.js(189:4) filters []`
^^^^^^^^^ This would be a minimum fix

*Ideally*
It would recognize index and pull the parent folder as such:
`Catalog/index.js(189:4) filters []`

**Comment by ZacharyCauchi** [comment]
Hi @kwelch, I might be interested in taking this issue on. I just want to know what the circumstances are that you want the parent directory to be included in the console statement? 

Thanks!

**Comment by kwelch** [comment]
I think only if the filename is index. I don't think the ext matters, but I am only aware of index resolving as when just the folder path is referred to so that should work.

**Comment by ZacharyCauchi** [comment]
Cheers. I'll try and get it done this weekend.

**Comment by kwelch** [comment]
@ZacharyCauchi How it is going? Did you have any questions or run into any issues I can help with?

**Comment by ZacharyCauchi** [comment]
@kwelch Hey mate. I took a look at the codebase, I thought this would be a simple trimming of the string thats logged with a regular expression - But I feel way out of depth and I've had a hard time finding the code that this issue pertains too 🙈. If you could point me in the right direction that'd be great but if you'd rather get it done quicker yourself its nbd

**Comment by kwelch** [comment]
@ZacharyCauchi 
Here is where the change would need to take place. https://github.com/kwelch/babel-plugin-captains-log/blob/master/src/index.js#L58-L66

You should be able to write a function (with tests) that takes my example file name above and call it with the `filename` variable before it is prepended to the arguments.

**Comment by ZacharyCauchi** [comment]
I tried running captains-log on some code I compiled with babel and I was able to get it to run - but I've been unable to set up a dev environment to work on the code. Ideally I'd be able to log the file object in the console to inspect it's contents and find the file path - but I'm unsure how to execute the index.js file I've edited on my local branch after cloning the repo. Any advice on how I can achieve this?

**Comment by kwelch** [comment]
Yeah, the bast way to run the index is to `npm link` this repo to another project and add it to your `.babelrc`. 

That being said, I typically use `npm run test` and verify all the tests pass with the updates I am looking for before attempting in a project.

In this case the filename that comes through the tests will not have the folder path as you would see when running in another app (this is why I missed it). 

If you want I can start a branch with the unit tests and leave it for you to put in the actual functionality.

**Comment by mntbkr** [comment]
Taking a stab at this, but I want to clarify the requirements....

> Ideally
It would recognize index and pull the parent folder as such:
Catalog/index.js(189:4) filters []

Is the ideal behavior to include the parent directory only when the file is index.js .... or any time we have a parent directory we include it plus the file name?

**Comment by kwelch** [comment]
Thinking more on this, I like the idea of it being based on the repo root in a super ideal scenario. 

Here are my preferences ranked:

Given this path: 
`/Users/welchk/_git/enga-web_new/packages/client/src/App/events/Views/Catalog/index.js`

Minimal: File name only
`index.js`

Nice to have: File name & parent `if filename === 'index.(ts|js|tsx|jsx)`
`Catalog/index.js`

Ideal: path from root of repo
`./packages/client/src/App/events/Views/Catalog/index.js`

**Comment by peteringram0** [comment]
Just incase anyone lands here who is using this plugin only for the console log file names and numbers at compile and is looking for this feature (waiting on this issue / PR) I have rolled my own version just for this: https://github.com/peteringram0/babel-plugin-console-source

**Comment by kwelch** [comment]
Nice! I was hoping to have more of the path, but I think jus the filename is a good start. My concern with just file name was a structure like `src/utils/index.js`, that `index.js` alone is not extremely helpful.

**Comment by peteringram0** [comment]
Ill add that to my one over the weekend, wont take long to add in.

**Comment by peteringram0** [comment]
Jut for anyone who needs it i added support for both file names and full file path: https://github.com/peteringram0/babel-plugin-console-source

**Comment by kwelch** [comment]
Awesome! I will probably back port that into this as well.

**Comment by peteringram0** [comment]
Your fully welcome to :)
